### PR TITLE
Update Dashboards plugins naming conventions

### LINKED ISSUE #47: Standardize plugin artifact names
Coming from https://github.com/opensearch-project/opensearch-plugins/issues/41#issuecomment-858139926

Now that Dashboards defaults to `kebab-case` for installed plugins, shall we finalize naming conventions for the build zips for Dashboards plugins? 

Current zip convention: `<pluginName>Dashboards-x.zip` (e.g., anomalyDetectionDashboards-1.0.0.0-rc1.zip`)

New ideas:
Option 1: identical to existing name, but in `kebab-case`: `<plugin-name>-dashboards-x.zip` (e.g., `anomaly-detection-dashboards-1.0.0.0-rc1.zip`.
Option 2: same convention as OpenSearch plugins, but replacing `opensearch` with `dashboards`: `dashboards-<plugin-name>-x.zip` (e.g., `dashboards-anomaly-detection-1.0.0.0-rc1.zip` (this was mentioned [here](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/322#issuecomment-840074136) as well)

Any thoughts here? I think either make sense. Option 1 will keep consistency with the existing plugin IDs which are following the camelcase convention of `<pluginNameDashboards>` (note that this could be changed to something like `dashboardsPluginName`, but will require more code changes & testing). Option 2 will keep consistency with OpenSearch plugin names, in regards to the zips at least.

### PR DESCRIPTION
**Pull request by ohltyler**
Signed-off-by: Tyler Ohlsen <ohltyler@amazon.com>

### Description
Updates CONVENTIONS.md regarding OpenSearch Dashboards plugin names examples. Removes issue that has since been resolved. Fixes `snake-case` -> `kebab-case`.
 
### Issues Resolved
#47 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by saratvemulapalli — DISMISSED** [review]
Thanks @ohltyler for the change!

**Comment by VachaShah** [comment]
@ohltyler The dashboards change for kebab-case [#322](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/322) is still pending review, do we want to wait before merging the conventions for Dashboards plugins?

**Comment by ohltyler** [comment]
@VachaShah I see, if there's risk of the merged changes being reverted, then we can leave a note about this. I'll add some wording back surrounding this.

**Review by saratvemulapalli — APPROVED** [review]

**Comment by VachaShah** [comment]
> @VachaShah I see, if there's risk of the merged changes being reverted, then we can leave a note about this. I'll add some wording back surrounding this.

Thank you @ohltyler!

**Review by VachaShah — APPROVED** [review]

**Comment by ohltyler** [comment]
Thanks for the reviews! I'm unable to merge, so please go ahead and merge yourselves :)

**Comment by saratvemulapalli** [comment]
> Thanks for the reviews! I'm unable to merge, so please go ahead and merge yourselves :)

I merged this and added you as a committer to this repository. 
You should now have all the permissions!

**Pull request merged**
